src/super_gradients/training/models/detection_models/customizable_detector.py

```diff
@@ -89,7 +107,71 @@ def __init__(
         self._default_multi_label_per_box = True
         self._default_class_agnostic_nms = False
 
+    def forward_sliding_window(self, images):
```

Comment:
IDK, I think this overcomplicate things.
What I was thinking about in my thoughts was an additional parameter to `predict` method that would enable tiled inference:
`model.predict(..., tiled=True)` while `tile_size` would be the default image size (we have this info already and `tile_step` would be equal to `tile_size` by default. Sure, user can override these params if he wants to. `tile=True` would be mutually exclusive with `skip_image_resizing` parameter for obvious reasons.

Moving this to `predict` would allow us to offer this feature to other OD models. WDYT?

Reply:
Putting it all in predict would make everything much easier of course.
On the other hand, this might come in handy when people want to use sliding window for producing benchmark results.